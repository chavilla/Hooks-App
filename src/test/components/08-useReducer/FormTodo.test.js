const { shallow } = require("enzyme");
const {
  default: FormTodo,
} = require("../../../components/08-useReducer/FormTodo");

describe("<FormTodo />", () => {
  const handleAddTodo = jest.fn();

  const wrapper = shallow(<FormTodo handleAddTodo={handleAddTodo} />);

  test("should to show succesfully", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("shouldnt to call handleAddTodo", () => {
      const formSubmit=wrapper.find('form').prop('onSubmit');

      formSubmit({preventDefault(){}});

      expect(handleAddTodo).toBeCalledTimes(0);

  });

  test('should to call handleAddTodo', () => {
    
    const value='Aprender Angular';

    wrapper.find('input').simulate('change', { target: { value, name: 'description' } } );

    wrapper.find('form').prop('onSubmit')({ preventDefault(){}});

    //si ha sido llamada
    expect(handleAddTodo).toHaveBeenCalled();

    //Si ha sido llamada con un objeto cualquiera
    expect(handleAddTodo).toHaveBeenCalledWith(expect.any(Object));

    //si ha sido llamada con un objeto en especifico
    expect(handleAddTodo).toHaveBeenCalledWith({
        desc: 'Aprender Angular',
        done:false,
        id: expect.any(Number)
    });
    
    //cuando la description ha sido vaciada
    expect(wrapper.find('input').prop('value')).toBe('');

  })
  

});
