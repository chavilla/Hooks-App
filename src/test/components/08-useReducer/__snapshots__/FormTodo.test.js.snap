// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormTodo /> should to show succesfully 1`] = `
<div
  className="col-5 d-flex flex-column justify-content-center"
>
  <h4>
    Agregar TODO
  </h4>
  <hr />
  <form
    onSubmit={[Function]}
  >
    <input
      autoComplete="off"
      className="form-control"
      name="description"
      onChange={[Function]}
      placeholder="Aprender..."
      type="text"
      value=""
    />
    <button
      className="btn btn-outline-primary mt-1 btn-block"
      type="submit"
    >
      Agregar
    </button>
  </form>
</div>
`;
